darwin/dataset/identifier.py

```diff
@@ -22,18 +26,15 @@ def __str__(self):
         return output
 
 
-def parse(slug: str):
-    try:
-        if "/" in slug:
-            team, slug = slug.split("/")
-        else:
-            team = None
-
-        if ":" in slug:
-            dataset, version = slug.split(":")
-        else:
-            dataset = slug
-            version = None
-        return team, dataset, version
-    except ValueError as e:
-        raise type(e)(f"Invalid dataset identifier {slug}")
+def _parse(slug: str) -> Tuple[str, str, Optional[str]]:
+    version: Optional[str] = None
+
+    if not re.fullmatch(r"[\w\-]+/[\w\-]+(:[\w\.]+)?", slug):
```

Comment:
For me, personally, this is hard to understand. Perhaps making this _explicit_ could be an option?

```
if not _is_dataset_valid(slug):
  raise ValueError(f"Invalid dataset identifier {slug}")
```
```